app/services/update_planned_disbursement.rb

```diff
@@ -8,6 +8,18 @@ def initialize(planned_disbursement:)
   def call(attributes: {})
     planned_disbursement.assign_attributes(attributes)
 
+    if attributes.key?(:financial_quarter) && attributes.key?(:financial_year)
+      planned_disbursement.period_start_date =
+        FinancialPeriod.start_date_from_quarter_and_year(attributes.fetch(:financial_quarter), attributes.fetch(:financial_year))
+      planned_disbursement.period_end_date =
+        FinancialPeriod.end_date_from_quarter_and_year(attributes.fetch(:financial_quarter), attributes.fetch(:financial_year))
+    elsif attributes.key?(:period_start_date) && attributes.key?(:period_end_date)
+      planned_disbursement.financial_quarter =
+        FinancialPeriod.quarter_from_date(attributes.fetch(:period_start_date).to_date)
+      planned_disbursement.financial_year =
+        FinancialPeriod.year_from_date(attributes.fetch(:period_start_date).to_date)
+    end
+
```

Comment:
I was wondering whether it could be worth wrapping this logic in a method or similar, since we seem to be repeating the same code in here and in the `create_planned_disbursement`. Is there a reason not to do it?

Reply:
I think I see what you are saying, something to make it clearer what is going on here? I am not sure how I would approach sharing anything between the two service objects though - do you have an approach in mind?

It's a bit naughty but the `elsif` block dissappears in the next PR I have lined up if that helps ease you mind?! :)

Reply:
Alright! That makes sense. I was surprised to find repeated code on your code because I know you are very thorough. I thought there would be a reason (and there is :-) ). 
I always struggle with scope. I still trip over things like thinking I have access to certain method or certain variable from a file just to find out that I don't. I wasn't sure you could easily share a method between services. Happy to leave it as it is then